### PR TITLE
feat: enables uds-runtime releases

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description

Adds everything necessary to deploy (besides CI, will add that in a separate PR, see https://github.com/defenseunicorns/uds-runtime/issues/96)

Publish with: `uds run build:publish-uds-runtime` 

uds-runtime image: `ghcr.io/defenseunicorns/uds-runtime:alpha`
uds-runtime Zarf pkg: `ghcr.io/defenseunicorns/packages/uds/uds-runtime:alpha`

Adds
- Dockerfile
- Helm chart including UDS Package
- load testing script (used to configure resource limits in chart)
- `zarf.yaml`
- tasks to support publishing all the things

Resolves https://github.com/defenseunicorns/uds-runtime/issues/89

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+replicaCount: 1
+image:
+  repository: ghcr.io/unclegedd/uds-runtime-dev
```
Comment (jeff-mccoy) [inline chart/values.yaml:3]: will have to update this when ready

File: zarf.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+metadata:
+  name: uds-runtime
+  description: tbd
+  version: 0.1.0
+
+components:
+  - name: uds-runtime
+    required: true
+    images:
+      - ghcr.io/unclegedd/uds-runtime-dev:dev
```
Comment (jeff-mccoy) [inline zarf.yaml:11]: same with this one

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Stage 1: Build Svelte frontend
+FROM node:21-alpine AS frontend-builder
```
Comment (jeff-mccoy) [inline Dockerfile:2]: can we switch all of these to chainguard images?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+# Stage 2: Build Go backend for amd64
+FROM golang:1.22.4-alpine AS backend-builder-amd64
+WORKDIR /app
+COPY go.mod go.sum ./
+RUN go mod download
+COPY . .
+COPY --from=frontend-builder /app/ui/build ./ui/build
+RUN CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -o /app/main-amd64 .
```
Comment (jeff-mccoy) [inline Dockerfile:15]: I dont think we need to build these parts inline, we can probably let our normal build stages create the binary and then set an arch target like we do with [zarf](https://github.com/zarf-dev/zarf/blob/main/Dockerfile)

File: Dockerfile (hunk 3 above)
Comment (UncleGedd) [reply Dockerfile]: done

File: Dockerfile (hunk 4 above)
Comment (UncleGedd) [reply Dockerfile]: 👍 set up tasks to do the build and then Dockerfile picks it up

File: chart/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - containerPort: 8080
+          resources:
+            requests:
+              memory: "128Mi"
+              cpu: "250m"
+            limits:
+              memory: "256Mi"
```
Comment (jeff-mccoy) [inline chart/templates/deployment.yaml:28]: Not sure how much data we have on resource usage yet, maybe we should bump the limits higher until we get that data?

File: chart/templates/uds-package.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        gateway: tenant
+        port: 8080
+        targetPort: 8080
+    allow:
+      - direction: Egress
+        selector:
+          app: uds-runtime
+        remoteGenerated: KubeAPI
+
+      - direction: Ingress
```
Comment (jeff-mccoy) [inline chart/templates/uds-package.yaml:22]: Do we need ingress here?

File: zarf.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+kind: ZarfPackageConfig
+metadata:
+  name: uds-runtime
+  description: tbd
+  version: '###ZARF_PKG_TMPL_REF###'
```
Comment (jeff-mccoy) [inline zarf.yaml:5]: We should be using release please or similar like uds code does. Package templates will eventually die.

**Review by jeff-mccoy — CHANGES_REQUESTED** [review]
Some comments

File: chart/templates/deployment.yaml (hunk 5 above)
Comment (UncleGedd) [reply chart/templates/deployment.yaml:28]: 👍 bumping limits

File: chart/templates/uds-package.yaml (hunk 6 above)
Comment (UncleGedd) [reply chart/templates/uds-package.yaml:22]: good catch, removing

File: zarf.yaml (hunk 7 above)
Comment (UncleGedd) [reply zarf.yaml]: updated to use release please

**Review by jeff-mccoy — APPROVED** [review]

**Pull request merged**